staging-service/src/main/java/com/github/cafdataprocessing/services/staging/DiskAccessHealthIndicatorWithTimeout.java

```diff
@@ -103,7 +99,15 @@ private synchronized void doWriteCheck(final Health.Builder builder) throws IOEx
             LOGGER.warn("Exception thrown trying to write healthcheck file to directory {} during healthcheck",
                         healthcheckFile.toString(), e);
         } finally {
-            Files.deleteIfExists(created);
```

Comment:
My thinking here was, the null pointer is being thrown obviously the files not being created, resulting in a null value for 'created', instead of having a check on the value at this point, wrap in a try catch similar to what was done with the healthcheckFile value, abstract the try catch to a new method to prevent duplication of code - will change to null check

Reply:
Ok, unfortunately we'd still have seen the NullPointerException because it wasn't being caught here.

        try {
            Files.deleteIfExists(path);
        } catch (final IOException e) {
            //Ignoring exception here due to wanting the healthcheck to only fail on the write check
        }

Reply:
This has been corrected in https://github.com/CAFDataProcessing/staging-service/pull/143/commits/eaa44970132f533db5857063115913409acb6d22